### PR TITLE
Drop ReadOnlyCollection dependency from TaskExceptionHolder

### PR DESCRIPTION
**Pull request by marek-safar**
it's used only to copy data aroud locally

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs [hunk 1]
```diff
@@ -277,17 +277,16 @@ internal AggregateException CreateExceptionObject(bool calledFromFinalizer, Exce
...
         /// </summary>
-        internal ReadOnlyCollection<ExceptionDispatchInfo> GetExceptionDispatchInfos()
+        internal List<ExceptionDispatchInfo> GetExceptionDispatchInfos()
         {
             List<ExceptionDispatchInfo>? exceptions = m_faultExceptions;
             Debug.Assert(exceptions != null, "Expected an initialized list.");
             Debug.Assert(exceptions.Count > 0, "Expected at least one exception.");
             MarkAsHandled(false);
-            return new ReadOnlyCollection<ExceptionDispatchInfo>(exceptions);
+            return exceptions!;
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs:289]: Why is the `!` needed here?

**Review by stephentoub — APPROVED** [review]

**Review by tarekgh — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs (hunk 1 above)
Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs:289]: The null check is done only inside Assert 3 lines above if that's what you mean

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs:289]: Yes.  The assert should mean that the `!` isn't needed.  Are you getting a compiler warning/error without the `!`?

Comment (marek-safar) [reply src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskExceptionHolder.cs]: Hmm, I think I was warning from it but cannot repro it again. Removing it

**Review by stephentoub — APPROVED** [review]

**Pull request merged**